.github/workflows/build.yml

```diff
@@ -0,0 +1,22 @@
+name: Build & Publish
+
+on: [ push ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: 8
+      - name: Build with Gradle
+        run: ./gradlew build
+      - name: Publish with Gradle
+        if: github.ref_type == 'tag'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TAG: ${{ github.ref_name }}
+        run: ./gradlew publish
```

Comment:
can you separate the jobs to test and publish?
so that test will run tests.
Publish will publish if tag is provided